### PR TITLE
Add `text` platform docs

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change

Adds docs for the new text platform.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/79454
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/3864
- Link to developer documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1536

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/text.markdown [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ha_quality_scale: internal
+ha_codeowners:
+  - '@home-assistant/core'
+  - '@raman325'
+ha_integration_type: entity
+---
+
+The Text integration is built for the controlling and monitoring of text values on devices.
+
+Text entities are created automatically by integrations that support them.
```
Comment (frenck) [inline source/_integrations/text.markdown]: This makes it a little more clear that you can't just spawn these manually, while also providing an alternative that can be used manually.

```suggestion
Text entities cannot be implemented manually, but can be provided by other integrations. If you are looking for a way to create a text entity, please take a look at the [Text helper](/integrations/input_text).
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

File: source/_integrations/text.markdown [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Text entities cannot be implemented manually, but can be provided by other integrations. If you are looking for a way to create a text entity, please take a look at the [Text helper](/integrations/input_text).
+
+## Services
+
+### text services
+
+Available services: `text.set_value`
+
+### Service `siren.turn_on`
```
Comment (frenck) [inline source/_integrations/text.markdown]:
```suggestion
### Service `text.set_value`
```

File: source/_integrations/text.markdown [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Turn the siren on.
```
Comment (frenck) [inline source/_integrations/text.markdown]:
```suggestion
Set the textual value of the text entity.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Not sure why I missed this earlier 🙈 

../Frenck

**Comment by raman325** [comment]
> Not sure why I missed this earlier 🙈
> 
> ../Frenck

doh me too

**Review by frenck — APPROVED** [review]
Thanks, @raman325 👍

**Pull request merged**